lib/src/layout/layout.dart

```diff
@@ -69,33 +76,177 @@ class _AgoraVideoViewerState extends State<AgoraVideoViewer> {
+                              .connectionData!.tokenUrl ==
+                          null
+                      ? ''
+                      : '&tokenUrl=${widget.client.sessionController.value.connectionData!.tokenUrl}';
+                  Share.share(
+                      'Please join my channel: "${widget.client.sessionController.value.connectionData!.channelName}" using this link: ${widget.client.deepLinkBaseUrl}/channelName=${widget.client.sessionController.value.connectionData!.channelName}' +
+                          tempTokenString +
+                          tokenUrlString);
+                },
+                child: Icon(
+                  Icons.copy,
+                  color: Colors.white,
+                  size: 24,
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    late String appId;
+    late String channelName;
+    String? tempToken;
+    String? tokenUrl;
+    // TODO: List<AreaCode>? areaCodes;
+    String? baseUrl;
+    String? layoutType = "grid";
+    Future<void> updateAgoraConnectionDataConfig(Object? link) async {
+      if (link != '') {
+        StreamSubscription<Uri?> subscription =
+            uriLinkStream.listen((Uri? uri) async {
+          print('uri: $uri');
+          baseUrl = uri?.origin;
+
+          uri?.queryParametersAll.forEach((key, value) {
+            print('params');
+            print('$key: ${value[0]}');
+            if (key == 'channelName') {
+              channelName = value[0];
+              print('Channe Name: $channelName');
+            } else if (key == 'tempToken') {
+              tempToken = value[0];
+              print('Temp Token: $tempToken');
+            } else if (key == 'tokenUrl') {
+              tokenUrl = value[0];
+              print('Token Url: $tokenUrl');
+            } else if (key == "layoutTpype") {
+              layoutType = value[0];
+              print("layoutType: $layoutType");
+            }
+          });
+        });
+
+        await subscription.asFuture();
+        final response = await http.get(
+          Uri.parse(
+            '$baseUrl/.well-known/assetlinks.json',
+          ),
+        );
+        if (response.statusCode == 200) {
+          final responseJson = jsonDecode(response.body);
+          appId = responseJson[0]['appId'];
+        } else {
+          print('response.statusCode: ${response.statusCode}');
+        }
+        await widget.client.sessionController.updateAgoraConnectionData(
+          appId: appId,
+          channelName: channelName,
+          tokenUrl: tokenUrl,
+          tempToken: tempToken,
+        );
+        await subscription.cancel();
+      }
+    }
+
+    return StreamBuilder(
+        stream: linkStream,
+        builder: (context, streamSnapshot) {
+          final link = streamSnapshot.data ?? '';
+          updateAgoraConnectionDataConfig(link);
+          if (link != '') {
```

Comment:
`if (!link.isEmpty) {`

Reply:
Link here is of object type, for which isEmpty is not valid

Reply:
I see. The method `updateAgoraConnectionDataConfig` accepts nil as an input, but if it receives nil it will do nothing if it receives nil or '', and it otherwise doesn't use the parameter anyway.
Seems like updateAgoraConnectionDataConfig should change to only allow a non-optional `Object`, remove the if statement inside that method and update this one to be:
```dart
if (streamSnapshot.data != null) {
    updateAgoraConnectionDataConfig(streamSnapshot.data);
    return userLayout(layoutType, baseUrl);
}
```